plugins/importexport/native/filter/NativeXmlSubmissionFilter.inc.php

```diff
@@ -205,15 +205,11 @@ public function &process(&$document)
         $importedObjects = & parent::process($document);
 
         // Index imported content
-        $submissionSearchIndex = Application::getSubmissionSearchIndex();
         foreach ($importedObjects as $submission) {
             assert($submission instanceof Submission);
-            $submissionSearchIndex->submissionMetadataChanged($submission);
-            $submissionSearchIndex->submissionFilesChanged($submission);
+            event(new MetadataChanged($submission));
         }
 
-        $submissionSearchIndex->submissionChangesFinished();
```

Comment:
The `submissionChangesFinished` function should be called after *all* submission files have been updated (in the case of a batch update); this is currently used by the Lucene plugin (and possibly others) to trigger re-indexing of content after a batch operation. I see you've moved it into the jobs that handle individual index changes (e.g. https://github.com/pkp/pkp-lib/pull/7049/files#diff-b954038ae0729b3fba218bfa9a9b05ad358a4e9b5104fdf01392be335cfbb24bR67). This will work fine for cases where only a single submission is changing, but in cases like this where a batch is being processed, it'll slow the indexing down considerably.

Reply:
Thank you for the explanation.

And I think, the unique job only works on Laravel 8.x. We are yet on the 7.x. I will check.